app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java

```diff
@@ -36,6 +38,14 @@
     private CesiumVectorLayersBridge mCesiumVectorLayersBridge;
     private CesiumMapBridge mCesiumMapBridge;
     private CesiumKMLBridge mCesiumKMLBridge;
+    private OnGGMapReadyListener mOnGGMapReadyListener;
+
+    /**
+     * A synchronized data holder is used to allow multi-threaded scenarios
+     * that occur while JS thread is updating data, that's simultaneously used
+     * by another thread
+     */
+    private SynchronizedDataHolder<Boolean> mIsGGMapReadySynchronized;
```

Comment:
why do you use this mIsGGMapReadySynchronized?
why dont use just callback. it will be much simpler.